### ISSUE TITLE
[JENKINS-272] workspace page delivers out-dated files

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#2266]
<p>I browse a workspace (in my case on a build slave) and view some generated<br/>
files. Then I re-build the project and try to look at the new generated files.</p>

<p>Expected: new contents<br/>
Experienced: old contents as seen when I first looked at it</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gradopado">gradopado</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272" target="_blank">workspace page delivers out-dated files</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2009-11-17T08:11:00+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I browse a workspace (in my case on a build slave) and view some generated
files. Then I re-build the project and try to look at the new generated files.</p>

<p>Expected: new contents
Experienced: old contents as seen when I first looked at it</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gradopado:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120437&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120437">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>very easy to proove: create a new job just with a single shell command:</p>

<p>date &gt; date.txt</p>

<p>In the worksapce view of this project, the date.txt file seems to be never updated.</p>
</pre>
</details>

</ul>
</details>

<p>very easy to proove: create a new job just with a single shell command:</p>

<p>date > date.txt</p>

<p>In the worksapce view of this project, the date.txt file seems to be never updated.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120439&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120439">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I tried to reproduce the problem, but it works correctly for me.</p>

<p>What browser are you using? Can you check the HTTP headers and tell me what it
reports? For Firefox, this extension would let you see the header:
<a href="http://livehttpheaders.mozdev.org/installation.html" class="external-link" target="_blank" rel="nofollow noopener">http://livehttpheaders.mozdev.org/installation.html</a></p>

<p>What I'm interested in is the headers like 
"If-Modified-Since: Thu, 08 Feb 2007 20:13:28 PST" that the browser is sending,
and  "Last-Modified: Thu, 08 Feb 2007 20:13:28 PST" that the server will send you.</p>
</pre>
</details>

</ul>
</details>

<p>I tried to reproduce the problem, but it works correctly for me.</p>

<p>What browser are you using? Can you check the HTTP headers and tell me what it<br/>
reports? For Firefox, this extension would let you see the header:<br/>
<a href="http://livehttpheaders.mozdev.org/installation.html" class="external-link" target="_blank" rel="nofollow noopener">http://livehttpheaders.mozdev.org/installation.html</a></p>

<p>What I'm interested in is the headers like <br/>
"If-Modified-Since: Thu, 08 Feb 2007 20:13:28 PST" that the browser is sending,<br/>
and  "Last-Modified: Thu, 08 Feb 2007 20:13:28 PST" that the server will send you.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gradopado:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120440&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120440">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>cache cleaned, first access to the file:
Last-Modified: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>



<ul>
	<li>page reload:
If-Modified-Since: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>


<p>HTTP/1.x 304 Not Modified</p>


<ul>
	<li>"build now" @ 	09.02.2007 10:58:42</li>
	<li>access file:
(old contents, no headers sent!)</li>
</ul>


<ul>
	<li>page reload:
If-Modified-Since: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>


<p>Last-Modified: Fri, 09 Feb 2007 10:58:42 CET</p>


<ul>
	<li>"build now" @ 09.02.2007 11:00:23</li>
	<li>access file:
(old contents, no headers sent!)</li>
</ul>


<ul>
	<li>page reload:
If-Modified-Since: Fri, 09 Feb 2007 10:58:42 CET</li>
</ul>


<p>Last-Modified: Fri, 09 Feb 2007 11:00:23 CET</p>


<ul>
	<li>page reload:
If-Modified-Since: Fri, 09 Feb 2007 11:00:23 CET</li>
</ul>


<p>HTTP/1.x 304 Not Modified</p>


<p>---------------------------
Firefox just doesn't ask for a new file, at least I did not see any header. (I'm
not an expert in HTTP.) A page reload seems to help. What do you think?</p>
</pre>
</details>

</ul>
</details>

<ul>
	<li>cache cleaned, first access to the file:<br/>
Last-Modified: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>



<ul>
	<li>page reload:<br/>
If-Modified-Since: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>


<p>HTTP/1.x 304 Not Modified</p>


<ul>
	<li>"build now" @ 	09.02.2007 10:58:42</li>
	<li>access file:<br/>
(old contents, no headers sent!)</li>
</ul>


<ul>
	<li>page reload:<br/>
If-Modified-Since: Fri, 09 Feb 2007 10:19:33 CET</li>
</ul>


<p>Last-Modified: Fri, 09 Feb 2007 10:58:42 CET</p>


<ul>
	<li>"build now" @ 09.02.2007 11:00:23</li>
	<li>access file:<br/>
(old contents, no headers sent!)</li>
</ul>


<ul>
	<li>page reload:<br/>
If-Modified-Since: Fri, 09 Feb 2007 10:58:42 CET</li>
</ul>


<p>Last-Modified: Fri, 09 Feb 2007 11:00:23 CET</p>


<ul>
	<li>page reload:<br/>
If-Modified-Since: Fri, 09 Feb 2007 11:00:23 CET</li>
</ul>


<p>HTTP/1.x 304 Not Modified</p>


<p>---------------------------<br/>
Firefox just doesn't ask for a new file, at least I did not see any header. (I'm<br/>
not an expert in HTTP.) A page reload seems to help. What do you think?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120441&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120441">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>OK, so the issue seems to be that Firefox doesn't even send a request unless you
hit F5.</p>

<p>Maybe I need to send another cache control header or something.</p>
</pre>
</details>

</ul>
</details>

<p>OK, so the issue seems to be that Firefox doesn't even send a request unless you<br/>
hit F5.</p>

<p>Maybe I need to send another cache control header or something.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120442&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120442">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Apparently when Last-Modified header is set but no Expires header is set,
Firefox automatically guesses the effective Expires header value and won't send
GET with If-Modified-Since until that effective Expires date is reached.</p>

<p>References:
<a href="http://labs.cybozu.co.jp/blog/kazuho/archives/2006/02/utilizing_cache.php" class="external-link" target="_blank" rel="nofollow noopener">http://labs.cybozu.co.jp/blog/kazuho/archives/2006/02/utilizing_cache.php</a>
<a href="http://www.mnot.net/cache_docs/" class="external-link" target="_blank" rel="nofollow noopener">http://www.mnot.net/cache_docs/</a>
(still looking for authoritative source)</p>

<p>The solution will be to send the Expires header with the same timestamp.</p>
</pre>
</details>

</ul>
</details>

<p>Apparently when Last-Modified header is set but no Expires header is set,<br/>
Firefox automatically guesses the effective Expires header value and won't send<br/>
GET with If-Modified-Since until that effective Expires date is reached.</p>

<p>References:<br/>
<a href="http://labs.cybozu.co.jp/blog/kazuho/archives/2006/02/utilizing_cache.php" class="external-link" target="_blank" rel="nofollow noopener">http://labs.cybozu.co.jp/blog/kazuho/archives/2006/02/utilizing_cache.php</a><br/>
<a href="http://www.mnot.net/cache_docs/" class="external-link" target="_blank" rel="nofollow noopener">http://www.mnot.net/cache_docs/</a><br/>
(still looking for authoritative source)</p>

<p>The solution will be to send the Expires header with the same timestamp.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120443&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120443">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fixed in 1.83.</p>
</pre>
</details>

</ul>
</details>

<p>Fixed in 1.83.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gradopado:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120444&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120444">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've set up a test job in hudson 1.83 which produces a file with a timestamp. It
should archive all artifacts (the file with timestamp).</p>

<p>The fix worked only once, then I still got stale files on the "latest artifacts"
page and on the workspace view also.</p>

<p>I'll check http headers if needed for your investigation.</p>
</pre>
</details>

</ul>
</details>

<p>I've set up a test job in hudson 1.83 which produces a file with a timestamp. It<br/>
should archive all artifacts (the file with timestamp).</p>

<p>The fix worked only once, then I still got stale files on the "latest artifacts"<br/>
page and on the workspace view also.</p>

<p>I'll check http headers if needed for your investigation.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120445&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120445">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 390 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 390 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-272?focusedId=120446&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-120446">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Bumping up the priority since another report was filed for this bug.</p>
</pre>
</details>

</ul>
</details>

<p>Bumping up the priority since another report was filed for this bug.</p>
